frontend/src/lib/api.ts

```diff
@@ -196,6 +204,32 @@ const api = {
         },
     },
 
+    pluginLogs: {
+        async search(
+            pluginConfigId: number,
+            currentTeamId: number | null,
+            searchTerm: string | null = null,
+            typeFilters: CheckboxValueType[] = [],
+            trailingEntry: PluginLogEntry | null = null,
+            leadingEntry: PluginLogEntry | null = null
+        ): Promise<PluginLogEntry[]> {
+            const type_filters =
+                typeFilters && typeFilters.length > 0 ? `&type_filter=${typeFilters.join('&type_filter=')}` : ''
+            const search = searchTerm ? `&search=${searchTerm}` : ''
+            const before = trailingEntry ? '&before=' + trailingEntry.timestamp : ''
+            const after = leadingEntry ? '&after=' + leadingEntry.timestamp : ''
+            const queryParams = `?limit=${LOGS_PORTION_LIMIT}${before}${after}${search}${type_filters}`
```

Comment:
Missed this in my first pass, but I think this is better if you create a POJO and then run it through `lib/utils.tsx/toParams`. As it is now all the params are also unsanitized, so for example searching for an URL probably won't work.